chromeExtension/src/components/StoreProvider.js

```diff
@@ -1,228 +1,242 @@
-                if (dynamicsState && dynamicsState.version < 9) { return; }
-
-                Honeybadger.notify(e, 'Failed to retrieve record ID while updating state');
-            }
-        }
-
-        const majorVersion = dynamicsState.version ?
-            parseInt(dynamicsState.version.split('.')[0], 10) : 0;
-
-        return {
-            context,
-            dynamicsState: {
-                isLoaded: true,
-                isForm,
-                recordId,
-                logicalName,
-                majorVersion,
-                ...dynamicsState
-            }
-        };
-    }
-    catch (e) {
-        Honeybadger.notify(e, {
-            message: 'Error encountered while determining isForm'
-        });
-    }
-}
-
-const StoreProvider = ({ children }) => {
-    const [state, dispatch] = useReducer(reducer, initialState);
-
-    const actions = useActions(state, dispatch);
-
-    const previousDynamicsState = useRef(defaultDynamicsState);
-
-    useEffect(() => {
-        let interval = setInterval(() => {
-            let { dynamicsState, context } = getDynamicsState();
-
-            if (!dynamicsState) {
-                return;
-            }
-
-            if (dynamicsState.fullVersion != previousDynamicsState.current.fullVersion ||
-                dynamicsState.isForm != previousDynamicsState.current.isForm ||
-                dynamicsState.recordId != previousDynamicsState.current.recordId ||
-                dynamicsState.logicalName != previousDynamicsState.current.logicalName) {
-
-                // Log the version
-                if (!previousDynamicsState.current.isLoaded && dynamicsState.fullVersion) {
-                    log(dynamicsState.fullVersion);
-                }
-
-                // Don't serialize the context (where `Xrm` lives)
-                // But DO cache it somewhere so it is easily accessible for debugging
-                window.__GOTDIBBS_TOOLBOX__.context = context;
-
-                // Only update the state if a key value has changed
-                actions.setDynamicsState(dynamicsState);
-                previousDynamicsState.current = dynamicsState;
-            }
-
-        }, 500);
-
-        return () => clearInterval(interval);
-    }, []);
-
-    return (
-        <StoreContext.Provider value={{ state, dispatch, actions }}>
-            {children}
-        </StoreContext.Provider>
-    );
-};
-
-export {
-    StoreContext,
-    StoreProvider
+import React, {
+    createContext,
+    useEffect,
+    useReducer,
+    useRef
+} from 'react';
+import Honeybadger from 'honeybadger-js';
+
+import { log } from '../logger';
+import { reducer, initialState } from '../store/reducer';
+import { useActions } from '../store/actions';
+
+const StoreContext = createContext(initialState);
+
+const defaultDynamicsState = {
+    isLoaded: false,
+    fullVersion: '0.0.0.0',
+    isForm: false,
+    recordId: null,
+    logicalName: null
+};
+
+const ignoreErrorPatterns = [

```

Comment:
Added this logic to reduce noise in the `if` statement that checks before doing any noisy error reporting below